### PR TITLE
Orders api

### PR DESCRIPTION
**Pull request by davyskiba**
This PR implements the second step of #38 - integrating with order API backend
When creating new classes went with desired naming (`order` instead of `trackable`) but didn't change it across the app as it is in the scope of #14

### CODE REVIEW

File: app/src/main/java/com/ably/tracking/demo/publisher/common/ListExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package com.ably.tracking.demo.publisher.common
+
+fun <T> List<T>.copyAndReplaceElementAt(index: Int, element: T) =
+    subList(0, index) + element + subList(index, size - 1)
```
Comment (KacperKluka) [inline app/src/main/java/com/ably/tracking/demo/publisher/common/ListExtensions.kt:4]: I'm not entirely sure what this function should do, but it seems to put a new element at the given index and remove the last element :thinking: Is it intentional?
```kotlin
val list = listOf(1, 2, 3, 4, 5, 6)
// (1, 2, 3, 4, 5, 6)
   
list.copyAndReplaceElementAt(2, 9)
// (1, 2, 9, 3, 4, 5)
```

File: app/src/main/java/com/ably/tracking/demo/publisher/domain/OrderManager.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+
+interface OrderManager {
+    val orders: MutableStateFlow<List<Order>>
```
Comment (KacperKluka) [inline app/src/main/java/com/ably/tracking/demo/publisher/domain/OrderManager.kt:22]: Is there a reason why it's not an immutable `StateFlow`? :thinking:

**Review by KacperKluka — CHANGES_REQUESTED** [review]
A few questions :wink:

File: app/src/main/java/com/ably/tracking/demo/publisher/common/ListExtensions.kt (hunk 1 above)
Comment (davyskiba) [reply app/src/main/java/com/ably/tracking/demo/publisher/common/ListExtensions.kt]: nice catch 🙇 
0257ff398ec19440ba78d6237762081cadc6fffc

File: app/src/main/java/com/ably/tracking/demo/publisher/domain/OrderManager.kt (hunk 2 above)
Comment (davyskiba) [reply app/src/main/java/com/ably/tracking/demo/publisher/domain/OrderManager.kt]: b5f8ba181aedd0a8aeff517733cfa2fe2d6dc992

File: app/src/main/java/com/ably/tracking/demo/publisher/common/ListExtensions.kt (hunk 1 above)
Comment (KacperKluka) [reply app/src/main/java/com/ably/tracking/demo/publisher/common/ListExtensions.kt]: Nice :+1: But now the name doesn't fit and I wonder what should be changed - the name or the implementation :thinking: Now it doesn't "copy and replace" but rather "copy and insert":
```kotlin
val list = listOf(1, 2, 3, 4, 5, 6)
// (1, 2, 3, 4, 5, 6)
   
list.copyAndReplaceElementAt(2, 9)
// (1, 2, 9, 3, 4, 5, 6)
```

Comment (davyskiba) [reply app/src/main/java/com/ably/tracking/demo/publisher/common/ListExtensions.kt]: eb2fd2957f436f77d7630caa45c72521da70fa0a

**Pull request merged**